backend/comballoc.ml

```diff
@@ -75,6 +75,15 @@ let rec combine i allocstate =
       let newnext = combine_restart i.next in
       (instr_cons_debug i.desc i.arg i.res i.dbg newnext,
        allocstate)
+  | Iop(Ibeginregion|Iendregion) -> begin
+      match allocstate with
+      | Pending_alloc { mode = Alloc_local; _ } ->
+          let newnext = combine_restart i.next in
+          (instr_cons_debug i.desc i.arg i.res i.dbg newnext, allocstate)
+      | No_alloc | Pending_alloc { mode = Alloc_heap; _ } ->
+          let newnext, s' = combine i.next allocstate in
+          (instr_cons_debug i.desc i.arg i.res i.dbg newnext, s')
+    end
   | Iop _ ->
```

Comment:
(Do we think we may have caught it earlier by
avoiding a catch-all pattern here?)

Reply:
Probably.  We should get warning 4 enabled.

Reply:
Yes, we should change that pattern.